src/cli/api/local_api.py

```diff
@@ -123,19 +127,122 @@ def list_applications(self) -> List[ApplicationType]:
+        - The json files in the config directory contain valid json
+        - The json files passes against schema validation
+
+        returns:
+            Returns empty list when all validations passes
+            Returns list containing error messages of the validations that failed
+        """
+        error_list = []
+        is_unique, _ = self.__is_application_unique(application)
 
-    def __is_config_valid(self, application: str) -> bool:
-        pass
+        if is_unique:
+            error_list.append("Application does not exist")
+        for items in self.__is_structure_valid(application):
+            error_list.append(items)
+        for items in self.__is_config_valid(application):
+            error_list.append(items)
+
+        # Remove any duplicates from list
+        filtered_error_list = []
+        for i in error_list:
+            if i not in filtered_error_list:
+                filtered_error_list.append(i)
+
+        return filtered_error_list
+
+    def __is_config_valid(self, application: str) -> List[str]:
+        # Validate if json string is correct and validate against json schema's
+        schema_app_path = Path(BASE_DIR + "/schema/applications")
+        app_config = Path(self.__config_manager.get_application(application)['path']) / "config"
+        application_path = app_config / "application.json"
+        network_path = app_config / "network.json"
+        result_path = app_config / "result.json"
+
+        error_list = []
+
+        # Validate application.json
+        if os.path.isfile(application_path):
+            json_valid, message = validate_json_string(application_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(application_path, schema_app_path / "appconfig_app.json")
+                if not schema_valid:
+                    error_list.append(ve)
+            else:
+                error_list.append(message)
+
+        if os.path.isfile(network_path):
+            json_valid, message = validate_json_string(network_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(network_path, schema_app_path / "appconfig_network.json")
+                if not schema_valid:
+                    error_list.append(ve)
+            else:
+                error_list.append(message)
+
+        if os.path.isfile(result_path):
+            json_valid, message = validate_json_string(result_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(result_path, schema_app_path / "appconfig_result.json")
+                if not schema_valid:
+                    error_list.append(ve)
+            else:
+                error_list.append(message)
+
+        return error_list
+
+    def __is_structure_valid(self, application_name: str) -> List[str]:
+        app_dir_path = Path(self.__config_manager.get_application(application_name)['path'])
+        app_config = app_dir_path / "config"
```

Comment:
I think we should name this directory app_config everywhere the same, check get_application_config where it is called app_config_path, in __create_application_structure it is called config_dir
Please make this consistent, also with the other instances of this path

The same accounts for / 'src' called app_dir or app_src etc.

Please check the other directories we use.